### ISSUE TITLE
fix: no-contradicting-variants flags necessary responsive resets

### ISSUE THREAD

**Issue by ftzi** [sergioazoc/oxlint-tailwindcss#150]
`no-contradicting-variants` reports `lg:block` as redundant in `block md:hidden lg:block`, although it is required to restore the element's display after `md:hidden`.

## Reproduction

Reproduced with `oxlint-tailwindcss@1.11.0`, `oxlint@1.75.0`, `tailwindcss@4.3.3`, and Node.js `24.18.0` on macOS. The published package's source revision matches the current `main` revision when checked.

Install the pinned packages in an empty directory with `npm install --save-dev oxlint@1.75.0 oxlint-tailwindcss@1.11.0 tailwindcss@4.3.3`.

`styles.css`

```css
@import "tailwindcss";
```

`.oxlintrc.json`

```json
{
  "jsPlugins": ["oxlint-tailwindcss"],
  "categories": { "correctness": "off" },
  "rules": {
    "tailwindcss/no-contradicting-variants": "error"
  },
  "settings": {
    "tailwindcss": { "entryPoint": "./styles.css" }
  }
}
```

`example.tsx`

```tsx
export const Example = () => <div className="block md:hidden lg:block" />;
```

Run `npx oxlint --config .oxlintrc.json example.tsx` from that directory.

## Actual behavior

```text
error tailwindcss(no-contradicting-variants): "lg:block" is redundant because "block" already applies unconditionally.
```

The CLI exits with status 1. A control containing `flex hover:flex` also reports, as expected.

## Expected behavior

No diagnostic for the responsive reset. With Tailwind's default breakpoints, the element is block below `48rem`, hidden from `48rem` to below `64rem`, and block again from `64rem` upward. Removing `lg:block` leaves it hidden at large widths.

This follows Tailwind's documented [mobile-first overrides](https://tailwindcss.com/docs/responsive-design#working-mobile-first). An unconditional base declaration is not necessarily the winning declaration at every breakpoint.

## Scope

This is distinct from #117. The reproduction uses a literal class list directly on a native `div`, with no merge helper or custom component. The current implementation checks whether the same utility exists in the base set but does not account for another variant overriding that property before the repeated utility restores it.

The rule should retain reports for genuinely redundant cases such as `flex hover:flex` while avoiding this responsive reset false positive.

## AI disclosure

The user requested this issue. GPT-6 Astra investigated the bug, ran the isolated CLI reproduction, and wrote this issue.

**Comment by sergioazoc** [comment]
Thanks for the report, @ftzi! 🙏 Really well put together: exact repro, pinned versions, and even the mobile-first cascade explanation. Wish every issue landed like this.

And yeah, you were right, it was a real false positive. The rule compared only by utility identity (`block` vs `block`) and had no notion of the responsive cascade, so it never saw that `md:hidden` overrides `display` in between. Dropping `lg:block` left the element hidden on large screens, exactly like you described.

The fix: before flagging a variant as redundant, the rule now checks whether another sibling class with a different utility writes an overlapping CSS property. If it does, it's a load-bearing reset and it stays quiet. With `entryPoint` configured it pulls the properties from the design system (covers any property, not just `display`), and without one it falls back to a static `display`/`visibility` set. It's strictly report-reducing, so the genuine cases like `flex hover:flex` or `flex dark:flex` still report the same.

Shipped in `oxlint-tailwindcss@1.11.1`. Already closed by the merge, but if you spot anything off feel free to reopen. Thanks again!
